sceptre/stack.py

```diff
@@ -638,10 +675,20 @@ def _wait_for_completion(self):
         self.most_recent_event_datetime = (
             datetime.now(tzutc()) - timedelta(seconds=3)
         )
-        while status == StackStatus.IN_PROGRESS:
-            status = self._get_simplified_status(self.get_status())
-            self._log_new_events()
-            time.sleep(4)
+        if timeout:
+            timeout = timeout * 60
+            elapsed = 0
+            while status == StackStatus.IN_PROGRESS and elapsed < timeout:
+                status = self._get_simplified_status(self.get_status())
+                self._log_new_events()
+                time.sleep(4)
+                elapsed += 4
+        else:
+            while status == StackStatus.IN_PROGRESS:
+                status = self._get_simplified_status(self.get_status())
+                self._log_new_events()
+                time.sleep(4)
```

Comment:
I'm not a fan of the duplicated code here; my suggestion is to add an inner function called `timed_out(elapsed)` function (which just returns `true` if no timeout is defined and otherwise checks whether elapsed is >= 60*timeout) and then have this block be:
```python
             elapsed = 0
             while status == StackStatus.IN_PROGRESS and not timed_out(elapsed):
                 status = self._get_simplified_status(self.get_status())
                 self._log_new_events()
                 time.sleep(4)
                 elapsed += 4
```
There is some redundant work for when there isn't a timeout, but it's a trivial amount and makes the code clearer.

Reply:
I like the idea. I assumed you wouldn't want inner functions for some reason. I did as you suggested (although the time_out function should return `False` if there's no timeout, not `True`)

Reply:
Yes, good call. I was too busy making sure I flipped the `>` that I overlooked the default behavior. 😆